plugin/service/storage/state_test.go

```diff
@@ -7,43 +7,41 @@ import (
 	"context"
 	"testing"
 
-	"github.com/aws/aws-sdk-go-v2/service/s3"
 	"github.com/hashicorp/boundary-plugin-aws/internal/credential"
-	"github.com/hashicorp/go-secure-stdlib/awsutil"
+	awsutilv2 "github.com/hashicorp/go-secure-stdlib/awsutil/v2"
 	"github.com/stretchr/testify/require"
 )
 
 func TestAwsStoragePersistedStateS3Client(t *testing.T) {
 	cases := []struct {
 		name             string
-		creds            *awsutil.CredentialsConfig
+		creds            *awsutilv2.CredentialsConfig
 		opts             []awsStoragePersistedStateOption
-		awsOpts          []awsutil.Option
+		awsOpts          []awsutilv2.Option
 		s3Opts           []s3Option
 		expectedRegion   string
 		expectedEndpoint string
 	}{
 		{
 			name: "static credentials",
-			creds: func() *awsutil.CredentialsConfig {
-				creds, err := awsutil.NewCredentialsConfig(
-					awsutil.WithRegion("us-west-2"),
-					awsutil.WithAccessKey("foobar"),
-					awsutil.WithSecretKey("barfoo"),
+			creds: func() *awsutilv2.CredentialsConfig {
+				creds, err := awsutilv2.NewCredentialsConfig(
+					awsutilv2.WithRegion("us-west-2"),
+					awsutilv2.WithAccessKey("foobar"),
+					awsutilv2.WithSecretKey("barfoo"),
 				)
 				require.NoError(t, err)
 				return creds
 			}(),
-			awsOpts: []awsutil.Option{
-				awsutil.WithIAMAPIFunc(
-					awsutil.NewMockIAM(),
+			awsOpts: []awsutilv2.Option{
+				awsutilv2.WithIAMAPIFunc(
+					awsutilv2.NewMockIAM(),
 				),
 			},
 			opts: []awsStoragePersistedStateOption{
 				withTestS3APIFunc(newTestMockS3(nil)),
 			},
 			s3Opts: []s3Option{
-				WithRegion("us-west-2"),
```

Comment:
does awsutil v2 use a default region?

Reply:
Yes, the default region is `us-east-1` here is a [link](https://github.com/hashicorp/go-secure-stdlib/blob/707a9aa19c93dbd028c3f29b25da8d35fd8ba4f6/awsutil/generate_credentials.go#L116) to the code.

Reply:
shouldn't this test be failing, then? since the expected region is `"us-west-2"`?

Reply:
The region option is set to `us-west-2` on line 29. The s3 options defined here are only for the s3 client, which does not include region.

Reply:
The expected region value is correct.